test/integration/routing.go

```diff
@@ -174,3 +193,48 @@ func (t *routing) verifyFaultInjection(src, dst, headerKey, headerVal string,
 	}
 	return nil
 }
+
+// verifyRedirect verifies if the http redirect was setup properly
+func (t *routing) verifyRedirect(src, dst, targetHost, targetPath, headerKey, headerVal string, respCode int) error {
+	url := fmt.Sprintf("http://%s/%s", dst, src)
+	glog.Infof("Making 1 request (%s) from %s...\n", url, src)
+
+	resp := t.clientRequest(src, url, 1, fmt.Sprintf("-key %s -val %s", headerKey, headerVal))
+
+	statusCode := ""
+	if len(resp.code) > 0 {
+		statusCode = resp.code[0]
+	}
+
+	if strconv.Itoa(respCode) != statusCode {
+		return fmt.Errorf("redirect verification failed: "+
+			"response status code: %v, expected %v",
+			statusCode, respCode)
+	}
+
+	exp := regexp.MustCompile("(?i)Host=(.*)")
+	hosts := exp.FindAllStringSubmatch(resp.body, -1)
```

Comment:
You should be able to make the logic a bit simpler since you only care about the first match:
https://golang.org/pkg/regexp/#Regexp.FindString
(or one of the ones right below it)